File: languages/java/src/test/java/IntegrityTest.java
import static org.junit.jupiter.api.Assertions.*;

import com.bloock.sdk.client.IntegrityClient;
import com.bloock.sdk.client.RecordClient;
import com.bloock.sdk.entity.integrity.Anchor;
import com.bloock.sdk.entity.integrity.Network;
import com.bloock.sdk.entity.integrity.Proof;
import com.bloock.sdk.entity.integrity.RecordReceipt;
import com.bloock.sdk.entity.record.Record;
import java.util.ArrayList;
import java.util.List;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

class IntegrityTest {

  @BeforeAll
  static void beforeAll() {
    Utils.initSdk();
  }

  @Test
  void integrityEndToEnd() throws Exception {
    ArrayList<Record> records = new ArrayList<>();

    RecordClient recordClient = new RecordClient();
    Record record = recordClient.fromString("Hello world").build();
    records.add(record);

    IntegrityClient integrityClient = new IntegrityClient();
    List<RecordReceipt> receipts = integrityClient.sendRecords(records);

    assertTrue(receipts.size() > 0);
    assertEquals(receipts.get(0).getRecord(), records.get(0).getHash());

    Anchor anchor = integrityClient.waitAnchor(receipts.get(0).getAnchor());

    assert receipts.get(0).getAnchor() == anchor.getId();

    Proof proof = integrityClient.getProof(records);

    String root = integrityClient.verifyProof(proof);
    assertNotSame("", root);
    assertNotNull(root);

    long timestampValidateRoot = integrityClient.validateRoot(root, Network.GNOSIS_CHAIN);

    assertTrue(timestampValidateRoot > 0);

    long timestampVerifyRecords = integrityClient.verifyRecords(records, Network.GNOSIS_CHAIN);

    assertTrue(timestampVerifyRecords > 0);
    assertEquals(timestampValidateRoot, timestampVerifyRecords);
  }
}

File: languages/php/tests/E2E/IntegrityTest.php
<?php

use Bloock\Bloock;
use Bloock\Client\IntegrityClient;
use Bloock\Client\RecordClient;
use Bloock\Entity\Integrity\Network;
use PHPUnit\Framework\TestCase;

final class IntegrityTest extends TestCase
{
    public static function setUpBeforeClass(): void
    {
        Bloock::$apiKey = getenv("API_KEY");
        Bloock::$apiHost = getenv("API_HOST");
    }

    public function testIntegrityEndToEnd()
    {
        $recordClient = new RecordClient();
        $record = $recordClient->fromString("Hello world")->build();
        $records = array($record);

        $integrityClient = new IntegrityClient();
        $receipts = $integrityClient->sendRecords($records);

        $this->assertNotEmpty($receipts);
        $this->assertEquals($receipts[0]->getRecord(), $records[0]->getHash());

        $anchor = $integrityClient->waitAnchor($receipts[0]->getAnchor());

        $this->assertEquals($receipts[0]->getAnchor(), $anchor->getId());

        $proof = $integrityClient->getProof($records);

        $root = $integrityClient->verifyProof($proof);

        $this->assertNotEquals("", $root);
        $this->assertNotNull($root);

        $timestampValidateRoot = $integrityClient->validateRoot($root, Network::GNOSIS_CHAIN);

        $this->assertTrue($timestampValidateRoot > 0);

        $timestampVerifyRecords = $integrityClient->verifyRecords($records, Network::GNOSIS_CHAIN);

        $this->assertTrue($timestampValidateRoot > 0);
        $this->assertEquals($timestampValidateRoot, $timestampVerifyRecords);
    }
}
